lib/config/store/pre-auth-hook.js

```diff
@@ -39,25 +39,25 @@ function onStorePreAuth (request, reply) {
     }
 
     return server.plugins.account.api.sessions.find(sessionToken)
-  })
 
-  .then(function (session) {
-    delete request.headers.authorization
-    request.headers.cookie = 'AuthSession=' + session.id
+    .then(function (session) {
```

Comment:
I agree that it’s not very readable, but in this case it has to be nested because of the `return` statements above. If we would chain the .then below outside of this `.then`, then it would be executed after the two `return` statements above and cause `reply` to be called twice